### ISSUE TITLE
Add ability to print long logs

### ISSUE THREAD

**Issue by pfives** [zserge/log#1]
Something simple like

```
    public static void longInfo(String str) {
        if(str.length() > 4000) {
            Log.i(TAG, str.substring(0, 4000));
            longInfo(str.substring(4000));
        } else
            Log.i(TAG, str);
    } }
```

would be really convenient for printing http responses and other long text

**Comment by Wavesonics** [comment]
That would be great!

**Comment by zserge** [comment]
Thanks for the request! I personally rarely use long logs, but I agree that this would be a nice feature.

I don't think it should be a separate method. Since all logs are coming through the `Log.print()` private function - the best way would be to split them there, so one could still use `Log.d` and friends and expect everything to be printed in 4K chunks.

Probably the easiest way would be to "steal" the algorithm from Timber (however your recusrive solution is really clever) - https://github.com/JakeWharton/timber/blob/master/timber/src/main/java/timber/log/Timber.java#L389 The advantage of timber's implementation is that is splits by newlines when possible. An improvement could to split on newlines and non-alphanumeric chars to not break words, yet to support single-line JSONs.

Also, I wonder if this should be done for both android log printer and System.out log printer? I don't think System.out has any limit, but I might be wrong.

**Comment by zserge** [comment]
I really wanted to adopt the recursive solution for its simplicity, but it seems like JVM crashes after ~8000 recursive calls, which corresponds to the log message of around 32MB long.

**Comment by zserge** [comment]
Ok, should be fixed now.
Long messages are wrapped on newline if possible, but if a single line of text is longer than 4000 bytes - it will be wrapped on the rightmost whitespace or punctuation.

https://github.com/zserge/log/commit/18241ab1b177672a7eb66c0b31050100c67f22c3

Can anyone confirm it in some real projects?

**Comment by zserge** [comment]
Covered my changes with tests, seems to be working.
Fixed in version 1.1.1. Closing this.
